lib/utils/compose_ts.ts

```diff
@@ -189,7 +189,7 @@ export async function tarDirectory(
 		pack.entry(
 			{
 				name: toPosixPath(fileStats.relPath),
-				size: fileStats.stats.size,
+				mtime: fileStats.stats.mtime,
```

Comment:
Removing `size`? I think that what we saw in the source code for `docker build` is that it gets the full header with "all fields":
* https://github.com/balena-os/balena-engine/blob/v19.03.11/pkg/archive/archive.go#L361
* https://golang.org/pkg/archive/tar/#FileInfoHeader

```
hdr, err := tar.FileInfoHeader(fi, link)
```
And then _modifies_ the full header, adjusting the following fields (and conditionally other fields elsewhere in source code):
https://github.com/balena-os/balena-engine/blob/v19.03.11/pkg/archive/archive.go#L366-L370
```
	hdr.ModTime = hdr.ModTime.Truncate(time.Second)
	hdr.AccessTime = time.Time{}
	hdr.ChangeTime = time.Time{}
	hdr.Mode = fillGo18FileTypeBits(int64(chmodTarEntry(os.FileMode(hdr.Mode))), fi)
	hdr.Name = canonicalTarName(name, fi.IsDir())
```
But, of course, we don't have to do the same that `docker build` does. :-)

Reply:
I just went for the smallest change, which was to add the mtime, everything else we can consider later (good catch on the size, missed it when reverting).